File: 0165_Compare Version Numbers/CompareVersionNumbers.cs
public class Solution {
    public int CompareVersion(string version1, string version2) {
        var ver1 = Convert(version1.Split('.'));
        var ver2 = Convert(version2.Split('.'));
        var m = 0;
        var n = 0;
        
        while(m < ver1.Count && n < ver2.Count)
        {
            if(ver1[m] > ver2[n]) return 1;
            else if(ver1[m] < ver2[n]) return -1;
            else{
                m++;
                n++;
            }
        }
        
        if(m < ver1.Count) return 1;
        if(n < ver2.Count) return -1;
        return 0;
        
    }
    
    private List<int> Convert(string[] splits)
    {
        var list = new List<int>();
        for(int i=splits.Length-1;i>=0;i--)
        {
            var v = int.Parse(splits[i]);
            if(v!=0 || (v == 0 && list.Count > 0)) list.Insert(0, v);
        }
        
        if(list.Count == 0) list.Add(0);
        return list;
    }
}

File: 0165_Compare Version Numbers/CompareVersionNumbers.java
class Solution {
    public int compareVersion(String version1, String version2) {
        String[] arrOfVer1 = version1.split("\\.");
        String[] arrOfVer2 = version2.split("\\.");
        int i = 0;
        while(i < Math.max(arrOfVer1.length, arrOfVer2.length)) {
            int v1 = i >= arrOfVer1.length ? 0 : Integer.parseInt(arrOfVer1[i]);
            int v2 = i >= arrOfVer2.length ? 0 : Integer.parseInt(arrOfVer2[i]);
            if(v1 < v2) return -1;
            else if(v1 > v2) return 1;
            i++;
        }

        return 0;
    }
}